### PR TITLE
Replace QValueKindAvroSchema with interface{}

### PR DESCRIPTION
**Pull request by serprex**
Instead of being either a string or a map[string]string,
create structs for the different avro schema definitions

Remove no longer applicable comments,
along with unused nullable parameter

### CODE REVIEW

File: flow/model/qvalue/avro_converter.go [hunk 1]
```diff
@@ -23,104 +35,64 @@ type QValueKindAvroSchema struct {
...
-			},
-		}, nil
+		return "string", nil
 	case QValueKindHStore, QValueKindJSON, QValueKindStruct:
-		return &QValueKindAvroSchema{
-			AvroLogicalSchema: map[string]interface{}{
-				"type":   "string",
-				"values": "string",
-			},
+		return AvroSchemaMap{
```
Comment (serprex) [inline flow/model/qvalue/avro_converter.go:64]: This is the only change in behavior, should be split out to a separate PR or reverted

Comment (serprex) [reply flow/model/qvalue/avro_converter.go:64]: https://github.com/PeerDB-io/peerdb/issues/847

**Review by iskakaushik — APPROVED** [review]
